pkg/blockdevice/zpool.go

```diff
@@ -113,15 +113,20 @@ func (z *Zpool) GetDevice() string {
 	return fmt.Sprintf("%s/%s", z.Name, z.DataSet)
 }
 
-func (z *Zpool) CheckFormatted() bool {
+func (z *Zpool) CheckFormatted() (bool, error) {
 	output, err := command.Run(fmt.Sprintf("zfs get -H lustre:fsname %s", z.GetDevice()), z.Log)
 	if err != nil {
-		return false
+		// If the error is because the data set doesn't exist yet, then that means it's not formatted
+		if strings.Contains(err.Error(), "dataset does not exist") {
+			return false, nil
+		}
+
+		return false, fmt.Errorf("could not run zfs to check for zpool device %w", err)
 	}
 
 	if len(output) == 0 {
-		return false
+		return false, fmt.Errorf("zfs get returned no output")
```

Comment:
You might consider quoting the `zfs get`, as it reads a bit funny without it: 

```suggestion
		return false, fmt.Errorf("'zfs get' returned no output")
```